src/coreclr/jit/llvm.cpp

```diff
@@ -864,31 +867,82 @@ Value* Llvm::consumeValue(GenTree* node, Type* targetLlvmType)
     return finalValue;
 }
 
+Function* Llvm::getOrCreateLlvmFunction(const char* symbolName, GenTreeCall* call)
+{
+    Function* llvmFunc = _module->getFunction(symbolName);
+
+    if (llvmFunc == nullptr)
+    {
+        // assume ExternalLinkage, if the function is defined in the clrjit module, then it is replaced and an
+        // extern added to the Ilc module
+        llvmFunc =
+            Function::Create(getFunctionTypeForCall(call), Function::ExternalLinkage, 0U, symbolName, _module);
+    }
+    return llvmFunc;
+}
+
 llvm::Value* Llvm::buildUserFuncCall(GenTreeCall* call)
 {
-    const char* symbolName = (*_getMangledSymbolName)(_thisPtr, call->gtEntryPoint.handle);
+    llvm::FunctionCallee llvmFuncCallee;
 
-    (*_addCodeReloc)(_thisPtr, call->gtEntryPoint.handle);
-    Function* llvmFunc = _module->getFunction(symbolName);
+    if (call->gtCallType == CT_USER_FUNC)
+    {
+        const char* symbolName = (*_getMangledSymbolName)(_thisPtr, call->gtEntryPoint.handle);
 
-    std::vector<llvm::Value*> argVec = std::vector<llvm::Value*>();
+        (*_addCodeReloc)(_thisPtr, call->gtEntryPoint.handle);
+        Function* llvmFunc = getOrCreateLlvmFunction(symbolName, call);
+        llvmFuncCallee = llvmFunc;
+    }
+    else if (call->gtCallType == CT_INDIRECT)
```

Comment:
```suggestion
    else if (call->gtCallType == CT_INDIRECT)
```
I would drop the `else if`, leaving only the `else`, and assert under it that the call is indirect.

This captures the idea that this method only builds 2 call types, and will only ever do that (well, it could eventually morph into just `buildCall`, but that'll be a different function).